src/settings/storeconf.cpp

```diff
@@ -9,126 +9,165 @@
-
-    if (json.contains("applicationName") && json["applicationName"].isString())
-        applicationName = json["applicationName"].toString();
-
-    if (json.contains("analizerStyle") && json["analizerStyle"].isString())
-        analizerStyle = json["analizerStyle"].toString();
-
-    if (json.contains("analizerFontSize") && json["analizerFontSize"].isString())
-        analizerFontSize = json["analizerFontSize"].toString();
-
-    if (json.contains("analizerFontName") && json["analizerFontName"].isString())
-        analizerFontName = json["analizerFontName"].toString();
-
-    if (json.contains("cppExtentions") && json["cppExtentions"].isString())
-        cppExtentions = json["cppExtentions"].toString();
+    QJsonObject json = mJsonDoc.object();
+    QMap<QString, QString>::const_iterator it = mFields.constBegin();
+    while (it != mFields.constEnd())
+    {
+        if (json.contains(it.key()) && json[it.key()].isString())   // overwrite default value
+            mFields[it.key()] = json[it.key()].toString();           // from json to QMap
+        ++it;
+    }
 
-    cppExtentionsList.clear();
+    mCppExtentionsList.clear();                                      // overwrite default value
 
-    std::string cppExtentionsS = cppExtentions.toStdString();
+    std::string cppExtentionsS = mFields["cppExtentions"].toStdString();
     std::vector<std::string> tokens;
     std::regex re("\\;+");
     std::sregex_token_iterator begin(cppExtentionsS.begin(), cppExtentionsS.end(), re, -1);
     std::sregex_token_iterator end;
     std::copy(begin, end, std::back_inserter(tokens));
     for (unsigned i = 0; i < tokens.size() ; i++)
-        cppExtentionsList << tokens[i].c_str();
+        mCppExtentionsList << tokens[i].c_str();
 }
 
 void StoreConf::saveData()
 {
-    QSettings settings(organizationName, applicationName);
-    settings.setValue("organizationName", organizationName);
-    settings.setValue("applicationVersion", applicationVersion);
-    settings.setValue("applicationName", applicationName);
-    settings.setValue("analizerStyle", analizerStyle);
-    settings.setValue("analizerFontSize", analizerFontSize);
-    settings.setValue("analizerFontName", analizerFontName);
-    settings.setValue("cppExtentions", cppExtentions);
-    settings.setValue("cppExtentionsList", QVariant::fromValue(cppExtentionsList));
-
-    QApplication::setOrganizationName(organizationName);
-    QApplication::setApplicationVersion(applicationVersion);
-    QApplication::setApplicationName(applicationName);
+    QApplication::setOrganizationName(mFields["organizationName"]);
+    QApplication::setApplicationVersion(mFields["applicationVersion"]);
+    QApplication::setApplicationName(mFields["applicationName"]);
+
+    QSettings settings;
+    QMap<QString, QString>::const_iterator it = mFields.constBegin();
+    while (it != mFields.constEnd())
+    {
+        settings.setValue(it.key(), mFields[it.key()]);        // copy values from QMap to QSettings
```

Comment:
What if there is already saved value?

Reply:
in start session user read value from settings
during session can change it couple times
method setValue() rewrite value